crates/shared/src/price_estimation/buffered.rs

```diff
@@ -52,13 +50,18 @@ pub trait NativePriceBatchFetching: Sync + Send + NativePriceEstimating {
     ///
     /// It returns a HashMap which maps the token with its native price
     /// estimator result
-    fn fetch_native_prices(
-        &self,
-        tokens: &HashSet<H160>,
+    fn fetch_native_prices<'a, 'b>(
+        &'a self,
+        tokens: &'b HashSet<H160>,
```

Comment:
To avoid annoying life times maybe we should pass this `HashSet` by value (probably isn't a performance isse in practice)?